### ISSUE TITLE
DoubleClick on rows Grid

### ISSUE THREAD

**Issue by ZiedHf** [telerik/kendo-react#215]
* Suggestion for improvement

## Current behavior
If we want to add a single click on the rows of Grid it could be this simple : `onRowClick={e => this.rowClick(e, 'onRowClick')}`
However if we want to add a **doubleClick** event on the rows it's a mess, this code would do the trick for **default Cell**. if we have another method to return **Custom Cell** we need to implement it there also : 

```
defaultCell(column) {
    return class DefaultCell extends GridCell {
      render() {
        const { dataItem, field } = this.props;
        let row;
        /**
        Logic to identify the row...
        **/
        return (
          <TableContext.Consumer>
            // Exemple with context
            {context => (
              <td
                onClick={e => {
                  e.preventDefault();
                  e.stopPropagation();
                  const { clicks, timeout } = context;
                  clicks.set(clicks.get() + 1); // Number of cliks
                  timeout.clearTimeout();
                  const data = { ...e, ...{ dataItem: dataItem } };
                  timeout.set(
                    setTimeout(() => {
                      if (clicks.get() > 1) context.onDoubleClick(dataItem); // Double Click if the number of _clicks_ more than 1 in time range less than _context.timeInMs_
                      else context.rowClick(data, 'onRowClick'); // Single Click
                      clicks.set(0); // Reset
                    }, context.timeInMs) // Time
                  );
                }}
                className={column.className}
              >
                {row} // the row data here
              </td>
            )}
          </TableContext.Consumer>
        );
      }
    };
  }
```

## Expected behavior
Just implementing doubleClick event.

## What is the motivation for improving the API of Grid ?
We're implementing a lot of code for the doubleClick event. Couldn't this be easier and an out-of-the-box feature or functionality. I mean it's very important to add a double click event for the Grid.
It would be nice also if it will be easy to add it to the _Custom Cells_.
My grid component is 900 lines of code. Almost 80 lines is just to add the `doubleClick` event to rows. A whole unnecessary new class for Default Cells and an unnecessary method to the Custom Cells to support it.

**Comment by Xizario** [comment]
@ZiedHf, If you are targeting latest browsers, they all support `dblclick` event for elements, so no more timeouts.
You can attach it using cell renderer. cell-renderers can be chained.

**Comment by ZiedHf** [comment]
@Xizario I have single click and double click on the same element.
This fires the two events.
And I don't want to.
https://jsfiddle.net/dLkfqyjg/
If there is another suggestion to achieve this it will be helpful.

**Comment by Xizario** [comment]
I am afraid in case you want to skip the logic of single click if double click is happening, there is no solution without a timeout. And this will delay the single click, which can be annoying for an end user. 
https://stackblitz.com/edit/react-wlmlax?file=app/main.js

From what I seen, most people prefer to execute the logic of a single click, and in case double click occurs to execute it as well. 
https://stackblitz.com/edit/react-wlmlax-1twcj7?file=app/main.js

**Comment by ZiedHf** [comment]
Anyway, It would be perfect if we can use only a props called **onDoubleClick** to avoid calling the **cellRender** method.

**Comment by vladimirivanoviliev** [comment]
Event added on our side, expect it in next dev versions.

**Comment by vladimirivanoviliev** [comment]
Available in: 3.7.0-dev.201911010953
